src/main/java/seedu/parser/NusModsParser.java

```diff
@@ -115,4 +117,20 @@ private static void addToFile(String input, String path) throws IOException {
         fw.close();
     }
 
+
+    public static Module fetchMod(String moduleCode) {
+        try {
+            String url = MODULE_API + moduleCode + ".json";
+            URL obj = new URL(url);
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+            con.setRequestMethod("GET");
+            con.setRequestProperty("User-Agent", "Mozilla/5.0");
+            JsonReader reader = new JsonReader(new InputStreamReader(con.getInputStream()));
+            return new Gson().fromJson(reader, Module.class);
+        } catch (Exception e) {
+            System.out.println("error");
+        }
+        return null;
+    }
+
```

Comment:
Is there another fetchMod method we can use that fetches from the downloaded json files instead of the API?

Reply:
Nope, there's another one that reads from the loaded modList but there isn't one that reads from the .json directly yet.